linsolve/linsolve.py

```diff
@@ -568,14 +612,18 @@ def __init__(self, data, wgts={}, sparse=False, **kwargs):
             c = np.log(constants[k]) # log unwraps complex circle at -pi
             logamp_consts[k], logphs_consts[k] = c.real, c.imag
         self.ls_amp = LinearSolver(logamp, logampw, sparse=sparse, constants=logamp_consts)
-        self.ls_phs = LinearSolver(logphs, logphsw, sparse=sparse, constants=logphs_consts)
+        if self.dtype in (np.complex64, np.complex128): # XXX worry about enumrating these explicitly
```

Comment:
This is a bit hacky, but what if we do `if np.iscomplexobj(np.zeros(1, dtype=self.dtype))`? There might be another way involve subtypes, but I'm not sure.

Reply:
I think this is best done as an overhaul of how data types are handled (which should be made explicit) in #33. Let's leave this as tbd for now. As implemented, I have not seen it break anything.